### PR TITLE
Add Snap support

### LINKED ISSUE #993: Feature Request: Create a Snap package
Originally started on https://github.com/abraunegg/onedrive/issues/853#issuecomment-633023826.

With the Snap packages, users would be able to use this client without having to manually compile it. And it would work in any Linux distribution.

### PR DESCRIPTION
**Pull request by felipecrs**
- [X] Make a working Snap

- [x] Claim name in Snap Store (https://github.com/abraunegg/onedrive/pull/994#discussion_r455541734)

- [x] Assure the build phase is optimized

- [ ] Request classic confinement in Snap Store ([forum post](https://forum.snapcraft.io/t/request-for-classic-confinement-for-onedrive/18984))

- [ ] Make completions work

- [ ] Expose `onedrive --monitor` as a service

- [ ] Create a GitHub Action in order to automate it

Closes #993.

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: onedrive # you probably want to 'snapcraft register <name>'
+base: core20 # the base snap is the execution environment for this snap
+version: '0.1' # just for humans, typically '1.2+git' or '1.3.2'
```
Comment (abraunegg) [inline snap/snapcraft.yaml:3]: Should this not be the version of the application?

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+summary: Single-line elevator pitch for your amazing snap # 79 char long summary
```
Comment (abraunegg) [inline snap/snapcraft.yaml:4]: Need a better summary

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+description: |
```
Comment (abraunegg) [inline snap/snapcraft.yaml:5]: Description needs to be updated

File: snap/snapcraft.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
+name: onedrive # you probably want to 'snapcraft register <name>'
```
Comment (abraunegg) [inline snap/snapcraft.yaml:1]: Has this been registered? Is there a conflict?

File: snap/snapcraft.yaml (hunk 1 above)
Comment (norbusan) [reply snap/snapcraft.yaml:3]: `2.4.3-snap1` or so I would suggest

File: snap/snapcraft.yaml (hunk 2 above)
Comment (norbusan) [reply snap/snapcraft.yaml:4]: `folder synchronization with OneDrive`

File: snap/snapcraft.yaml (hunk 3 above)
Comment (norbusan) [reply snap/snapcraft.yaml:5]:
```
 OneDrive is the cloud storage system of Microsoft. This package provides
 the command line client specialising in synchronizing with OneDrive cloud
 storage.
```

**Comment by felipecassiors** [comment]
I'm sorry @abraunegg, this is a draft PR and I forgot to say that's is not ready to review. But thank you so much for your time anyway, I'll address the way you suggested.

File: snap/snapcraft.yaml (hunk 4 above)
Comment (felipecassiors) [reply snap/snapcraft.yaml:1]: Yes, it was previously registered. But it was never published. We'll have to send a request to claim this name.

Comment (felipecassiors) [reply snap/snapcraft.yaml:1]: I tried to fill a name claim, but I was not able to due to [this bug](https://bugs.launchpad.net/snapcraft/+bug/1771767).

Comment (felipecassiors) [reply snap/snapcraft.yaml:1]: In the worst case, we can use "onedrive-abraunegg".

Comment (felipecassiors) [reply snap/snapcraft.yaml]: Ok, I successfully opened a name claim, let's wait to see what they say.

PS: I used the following URL as pointed [here](https://bugs.launchpad.net/snapstore-server/+bug/1771767/comments/7).

https://snapcraft.io/register-name-dispute?snap-name=onedrive

Comment (felipecassiors) [reply snap/snapcraft.yaml]: Great news, I successfully claimed the `onedrive` name in Snap Store. Feel free to tell me if you happen to want me to transfer the ownership to you.

Comment (abraunegg) [reply snap/snapcraft.yaml]: Thats great you have claimed the name, but please - you can retain ownership here for that.

**Comment by abraunegg** [comment]
@felipecrs 
Any update on this PR ?

**Comment by felipecrs** [comment]
@abraunegg not really, sorry.

The latest news is that [the guys from Snap Store did not agree about giving permission for classic confinement](https://forum.snapcraft.io/t/request-for-classic-confinement-for-onedrive/18984). Because of that, we will have to set all the permissions accordingly, which is probably the trickiest part.

I'm not with so much extra time these days, so I can't give an ETA for this. If anyone wants to continue the work, I'll be glad to help. Currently, I own the `onedrive` namespace on the Snap Store.

**Comment by abraunegg** [comment]
@felipecrs 
Closing this pull request for the following reason:
* Blocked by Snap Team for folder / permissions issue

**Pull request closed without merging**